perfkitbenchmarker/linux_benchmarks/iperf_benchmark.py

```diff
@@ -171,74 +184,190 @@ def _RunIperf(sending_vm, receiving_vm, receiving_ip_address, thread_count,
+      interval_start_time_list = []
+      rtt_avg_list = []
+      cwnd_avg_list = []
+      netpwr_sum_list = []
+      retry_sum_list = []
+      cwnd_scale = None
+      total_stats = {}
+
+      if thread_count > 1:
+        # Parse aggregates of multiple sending threads for each report interval
+        r = re.compile((
+            r'\[SUM\]\s+(?P<interval>\d+\.\d+-\d+\.\d+)\s\w+\s+(?P<transfer>\d+\.?\d*)'
+            r'\s\w+\s+(?P<throughput>\d+\.?\d*)\sMbits\/sec\s+(?P<write>\d+)'
+            r'\/(?P<err>\d+)\s+(?P<retry>\d+)'))
+        interval_sums = [m.groupdict() for m in r.finditer(stdout)]
+
+        # Parse output for each individual thread for each report interval
+        r = re.compile((
+            r'\[\s*(?P<thread_num>\d+)\]\s+(?P<interval>\d+\.\d+-\d+\.\d+)\s+sec\s+'
+            r'(?P<transfer>\d+\.?\d*)\s\w+\s+(?P<throughput>\d+\.?\d*)\sMbits\/sec\s+(?P<write>\d+)\/'
+            r'(?P<err>\d+)\s+(?P<retry>\d+)\s+(?P<cwnd>-?\d+)(?P<cwnd_scale>\w*)\/(?P<rtt>\d+)\s+'
+            r'(?P<rtt_unit>\w+)\s+(?P<netpwr>\d+\.?\d*)'))
+        interval_threads = [m.groupdict() for m in r.finditer(stdout)]
+
+        # sum and average across threads for each interval report
+        for interval in interval_sums[:-1]:
+          interval_time = interval['interval'].split('-')
+          interval_start = float(interval_time[0])
+          interval_start_time_list.append(interval_start)
+          interval_throughput_list.append(float(interval['throughput']))
+
+          thread_results_for_interval = list(filter(lambda x: x['interval'] == interval['interval'], interval_threads))
+          if len(thread_results_for_interval) != thread_count:
+            logging.warning("iperf thread results don't match sending_thread argument")
+
+          rtt_sum = 0.0
+          cwnd_sum = 0.0
+          netpwr_sum = 0.0
+          retry_sum = 0
+
+          for thread_result in thread_results_for_interval:
+            rtt_sum += float(thread_result['rtt'])
+            cwnd_sum += float(thread_result['cwnd'])
+            netpwr_sum += float(thread_result['netpwr'])
+            retry_sum += int(thread_result['retry'])
+            cwnd_scale = thread_result['cwnd_scale']
+
+          rtt_average = rtt_sum/len(thread_results_for_interval)
+          cwnd_average = cwnd_sum/len(thread_results_for_interval)
+
+          rtt_avg_list.append(rtt_average)
+          cwnd_avg_list.append(cwnd_average)
+          netpwr_sum_list.append(netpwr_sum)
+          retry_sum_list.append(retry_sum)
+        
+        total_stats = interval_sums[len(interval_sums)-1]
+        total_stats['rtt'] = sum(rtt_avg_list)/len(rtt_avg_list)
+        total_stats['cwnd'] = sum(cwnd_avg_list)/len(cwnd_avg_list)
+        total_stats['netpwr'] = sum(netpwr_sum_list)/len(netpwr_sum_list)
+        total_stats['rtt_unit'] = thread_results_for_interval[0]['rtt_unit']
+
+        total_throughput = total_stats['throughput']
+
+      elif thread_count == 1:
```

Comment:
I suggest adding a check for thread_count < 1, so that the error won't occur when creating tcp_metadata.

Reply:
I have added a check in _Run to ignore thread_count that are less than 1